### PR TITLE
SOC-2278 | Change discussion header color if themeDesigner button color is below certain treshold

### PR DESCRIPTION
**Pull request by Bloodmoon24**
## Links

* https://wikia-inc.atlassian.net/browse/SOC-2278

## Reviewers
@kvas-damian  @bkoval

### CODE REVIEW

File: front/main/app/mixins/theme.js [hunk 1]
```diff
@@ -51,12 +51,32 @@ export default Ember.Mixin.create(ColorUtilsMixin, {
 	},
 
 	/**
+	 * Changes discussion header color if HSL Lightness is above 0.7
+	 * @param {object} rgb
```
Comment (bkoval) [inline front/main/app/mixins/theme.js]: Object should be capitalized. I think we're changing this everywhere on discussions to live up to Mercury's standards.

File: front/main/app/mixins/theme.js [hunk 2]
```diff
@@ -69,11 +89,15 @@ export default Ember.Mixin.create(ColorUtilsMixin, {
...
 
+		heroImageRgbColor = this.hexToRgb(this.get('themeColors.color-buttons'), 0.8);
+		discussionHeaderColor = this.getHeaderColor(heroImageRgbColor);
+
 		styles += `.discussions .border-theme-color {border-color: ${this.get('themeColors.color-buttons')};}`;
 		styles += `.discussions .background-theme-color {background-color: ${this.get('themeColors.color-buttons')};}`;
-		styles += '.discussions .background-alpha-theme-color {background-color: ' +
-			`${this.getRgbaColor(this.hexToRgb(this.get('themeColors.color-buttons'), 0.8))};}`;
-
+		styles += `.discussions .background-alpha-theme-color {background-color: ${this.getRgbaColor(heroImageRgbColor)};}`;
```
Comment (bkoval) [inline front/main/app/mixins/theme.js]: Doesn't it need RGBA with opacity anymore?

File: front/main/app/mixins/theme.js [hunk 3]
```diff
@@ -51,12 +51,32 @@ export default Ember.Mixin.create(ColorUtilsMixin, {
 	},
 
 	/**
+	 * Changes discussion header color if HSL Lightness is above 0.7
+	 * @param {object} rgb
+	 * @returns {string}
+	 */
+	getHeaderColor(rgb) {
+		const defaultColor = '#FFFFFF',
```
Comment (bkoval) [inline front/main/app/mixins/theme.js]: How about we use lower case for hex as we do in the stylesheets?

**Comment by bkoval** [comment]
This is looking very good. Thanks!

File: front/main/app/mixins/theme.js (hunk 2 above)
Comment (Bloodmoon24) [inline front/main/app/mixins/theme.js]: Yeah, it's still sing opacity ;) https://github.com/Wikia/mercury/pull/2410/files#diff-e664976bef7532bb09bcb36e2f0fa459R92

File: front/main/app/mixins/theme.js (hunk 3 above)
Comment (Bloodmoon24) [inline front/main/app/mixins/theme.js]: Thanks for that

File: front/main/app/mixins/theme.js [hunk 4]
```diff
@@ -51,12 +50,26 @@ export default Ember.Mixin.create(ColorUtilsMixin, {
...
+	 * Changes discussion header color if HSL Lightness is above 0.7
+	 * @param {Object} color
+	 * @returns {string}
+	 */
+	getHeaderColor(color) {
+		const defaultColor = '#ffffff',
+			fallbackAboveLightness = 0.7,
+			fallbackColor = '#1a1a1a';
+
+		return (color.getLighness() < fallbackAboveLightness) ? defaultColor : fallbackColor;
```
Comment (kvas-damian) [inline front/main/app/mixins/theme.js]: I'd suggest using class names instead of color directly. This way we can get all styling in one place - SCSS.

Comment (Bloodmoon24) [inline front/main/app/mixins/theme.js]: Yeah, but then you will need to pass the color to three components, which in my opinion is more complicated

File: front/main/app/mixins/theme.js [hunk 5]
```diff
@@ -51,12 +51,26 @@ export default Ember.Mixin.create(ColorUtilsMixin, {
 	},
 
 	/**
+	 * Changes discussion header color if HSL Lightness is above 0.7
+	 * @param {Object} color
+	 * @returns {string}
+	 */
+	getHeaderColor(color) {
+		const defaultColor = '#ffffff',
+			fallbackAboveLightness = 0.7,
```
Comment (kvas-damian) [inline front/main/app/mixins/theme.js]: Maybe we could rename it to `lightnessThreshold`?

File: front/main/app/mixins/theme.js [hunk 6]
```diff
@@ -69,11 +83,15 @@ export default Ember.Mixin.create(ColorUtilsMixin, {
...
+
 		styles += `.discussions .border-theme-color {border-color: ${this.get('themeColors.color-buttons')};}`;
 		styles += `.discussions .background-theme-color {background-color: ${this.get('themeColors.color-buttons')};}`;
-		styles += '.discussions .background-alpha-theme-color {background-color: ' +
-			`${this.getRgbaColor(this.hexToRgb(this.get('themeColors.color-buttons'), 0.8))};}`;
-
+		styles += `.discussions .background-alpha-theme-color {background-color: ${heroImageRgbColor.toRgbString()};}`;
+		styles += `.discussions .discussion-hero-unit .discussion-hero-unit-content h1 {color: ${discussionHeaderColor};}`;
+		styles += `.discussions .discussion-hero-unit .discussion-hero-unit-content p {color: ${discussionHeaderColor};}`;
+		styles += `.discussion-header h1 {color: ${discussionHeaderColor};}`;
```
Comment (kvas-damian) [inline front/main/app/mixins/theme.js]: please merge those styles into one selector ;)

**Comment by kvas-damian** [comment]
LGTM

**Comment by bkoval** [comment]
LGTM 🔵

**Pull request merged**